### PR TITLE
Allow using arrays as input to `alt`

### PR DESCRIPTION
**Pull request by glittershark**
Now that min-const-generics is stabilized, it's possible to (backwards-compatibly) write a version of `alt` that uses a const array as input instead of being limited to only tuples - this allows us to lift the 21-element limitation on tuples passed to `alt`.

### CODE REVIEW

**Comment by MTRNord** [comment]
Hi,

any update on this? I am running into a case where I have a grammar with 35 cases. Obviously this doesnt't work with current nom

**Comment by Geal** [comment]
there were a few things missing to make the DX on this reasonable, mainly having most parsers return a concrete type instead of a closure, and having a `Either` parser. All elements of the array have to be of the same type. Now we can revisit this :)

**Review by Geal — APPROVED** [review]

**Comment by Geal** [comment]
great work @glittershark, thanks!

File: src/branch/mod.rs [hunk 1]
```diff
@@ -178,22 +178,34 @@ impl<
...
+    &mut self,
+    input: Input,
+  ) -> crate::PResult<OM, Input, Self::Output, Self::Error> {
     let mut error = None;
 
-    for branch in self {
-      match branch.parse(input.clone()) {
+    for branch in &mut self.parser {
+      match branch.process::<OM>(input.clone()) {
+        //branch.parse(input.clone()) {
```
Comment (glittershark) [inline src/branch/mod.rs:196]: should this have been committed?

**Comment by Geal** [comment]
good catch, fixed in 8c68e22

**Pull request merged**